File: src/dere_plugin/skills/recalling-past-context/scripts/session_history.py
#!/usr/bin/env python3
"""Get session conversation history."""

from __future__ import annotations

import json
import sys
from pathlib import Path

import requests

sys.path.insert(0, str(Path(__file__).parent.parent.parent / "scripts"))
from config_reader import get_daemon_url


def get_session_history(session_id: int) -> dict | None:
    """Get conversation history for session."""
    daemon_url = get_daemon_url()

    try:
        response = requests.get(
            f"{daemon_url}/sessions/{session_id}/history",
            timeout=5
        )
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error getting session history: {e}", file=sys.stderr)
        return None


def main():
    """Main entry point."""
    if len(sys.argv) < 2:
        print("Usage: session_history.py <session_id>", file=sys.stderr)
        sys.exit(1)

    try:
        session_id = int(sys.argv[1])
    except ValueError:
        print("Error: session_id must be an integer", file=sys.stderr)
        sys.exit(1)

    result = get_session_history(session_id)

    if result:
        print(json.dumps(result, indent=2))
    else:
        sys.exit(1)


if __name__ == "__main__":
    main()

File: plugins/dere_core/skills/recall/scripts/session_history.ts
import { getDaemonUrl } from "../../../scripts/config_reader.js";

async function getSessionHistory(sessionId: number): Promise<Record<string, unknown> | null> {
  const daemonUrl = await getDaemonUrl();
  const controller = new AbortController();
  const timeout = setTimeout(() => controller.abort(), 5_000);
  try {
    const response = await fetch(`${daemonUrl}/sessions/${sessionId}/history`, {
      method: "GET",
      signal: controller.signal,
    });
    if (!response.ok) {
      console.error(`Error getting session history: ${response.status} ${response.statusText}`);
      return null;
    }
    return (await response.json()) as Record<string, unknown>;
  } catch (error) {
    console.error(`Error getting session history: ${String(error)}`);
    return null;
  } finally {
    clearTimeout(timeout);
  }
}

async function main(): Promise<void> {
  const sessionIdArg = process.argv[2];
  if (!sessionIdArg) {
    console.error("Usage: session_history.ts <session_id>");
    process.exit(1);
  }

  const sessionId = Number.parseInt(sessionIdArg, 10);
  if (!Number.isFinite(sessionId)) {
    console.error("Error: session_id must be an integer");
    process.exit(1);
  }

  const result = await getSessionHistory(sessionId);
  if (result) {
    console.log(JSON.stringify(result, null, 2));
  } else {
    process.exit(1);
  }
}

if (import.meta.main) {
  void main();
}
